### PR TITLE
[serve] Fix whacky worker replica failure test

### PR DESCRIPTION
**Pull request by edoakes**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/tests/test_failure.py [hunk 1]
```diff
@@ -157,44 +155,45 @@ def __call__(self, *args):
...
-    while len(responses) == 1:
-        responses.add(request_with_retries("/replica_failure", timeout=1).text)
+    start = time.time()
+    while time.time() - start < 30:
         time.sleep(0.1)
+        responses.add(request_with_retries("/replica_failure", timeout=1).text)
+        if len(responses) > 1:
+            break
+    else:
+        raise TimeoutError("Timed out waiting for replicas after 30s.")
```
Comment (ijrsvt) [inline python/ray/serve/tests/test_failure.py:197]: @edoakes Should we actually check that we get some response?

Comment (edoakes) [reply python/ray/serve/tests/test_failure.py:197]: good call, let me add an assert

**Comment by edoakes** [comment]
@ijrsvt comment addressed

**Comment by ijrsvt** [comment]
> @ijrsvt comment addressed

Did you push?

**Comment by edoakes** [comment]
@ijrsvt oops, pushed and looked away without checking that it succeeded :) updated now

**Review by ijrsvt — COMMENTED** [review]
@edoakes Looks good, but I was also wondering if you needed to check the return value on line 209?

**Comment by edoakes** [comment]
@ijrsvt we could do that, but I don't think it's necessary and is a bit messy because it'd require adding some code to wait for the other replica to crash and restart. The first couple of requests could in theory be handled by the `kill`ed replica.

**Review by ijrsvt — COMMENTED** [review]
LGTM!

**Review by ijrsvt — APPROVED** [review]

**Pull request merged**